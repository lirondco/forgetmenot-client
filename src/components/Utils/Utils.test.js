import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Hyph, Button, Textarea, Input, Required, Section } from './Utils'

describe(`Utils components`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  }

  it('renders a hyphen when Hyph is called', () => {
    const wrapper = shallow(<Hyph />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('button renders when called', () => {
    const wrapper = shallow(<Button {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('button works when clicked', () => {
      const wrapper = shallow(<Button {...props} />)
      wrapper.find('button').simulate('click', { preventDefault() {}})
  })

  it('textarea renders when called', () => {
      const wrapper = shallow(<Textarea {...props} />)
      expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('input renders when called', () => {
      const wrapper = shallow(<Input {...props} />)
      expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('required renders when called', () => {
      const wrapper = shallow(<Required {...props} />)
      expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('section renders when called', () => {
      const wrapper = shallow(<Section {...props} />)
      expect(toJson(wrapper)).toMatchSnapshot()
  })
})
