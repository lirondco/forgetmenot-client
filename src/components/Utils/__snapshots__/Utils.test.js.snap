// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils components button renders when called 1`] = `
<button
  className="Button test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</button>
`;

exports[`Utils components input renders when called 1`] = `
<input
  className="Input test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</input>
`;

exports[`Utils components renders a hyphen when Hyph is called 1`] = `
<span
  className="Hyph"
>
   - 
</span>
`;

exports[`Utils components required renders when called 1`] = `
<span
  className="Required test-class-name"
  data-other="test-other-prop"
>
  *
</span>
`;

exports[`Utils components section renders when called 1`] = `
<section
  className="Section test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</section>
`;

exports[`Utils components textarea renders when called 1`] = `
<textarea
  className="Textarea test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</textarea>
`;
